### PR TITLE
Initial Interfaces

### PR DESCRIPTION
**Pull request by mecha**
The initial interfaces.

### CODE REVIEW

**Comment by mecha** [comment]
@XedinUnknown What do you think of the namespace `Dhii\Util\Invocation`, considering that `StringableInterface` which is similar is in `Dhii\Util\String`?

**Review by XedinUnknown — APPROVED** [review]

File: src/InvocableInterface.php [hunk 1]
```diff
@@ -10,13 +10,17 @@
...
-     * This method will be run when the instance is called as a function.
+     * Invokes this instance as a function.
      *
      * For instance:
      * If `$obj` is an instance of the implementing class, doing `$obj()` will invoke this method.
      * That pun was intended.
      *
      * @since [*next-version*]
+     *
+     * @param mixed $argN An argument to the invocation.
```
Comment (mecha) [inline src/InvocableInterface.php]: Should be "Any number of arguments", no?

Comment (XedinUnknown) [reply src/InvocableInterface.php]: That would kinda hint that this is a set or array.

**Comment by XedinUnknown** [comment]
@mecha, this is now ready for review.

File: src/InvocableInterface.php (hunk 1 above)
Comment (mecha) [reply src/InvocableInterface.php]: Then we need a better way to document variable number of parameters. PSR-5 is missing this information. ApiGen and PHPDocumenter use this format:

```
@param type $name,... The description.
```

**Review by mecha — COMMENTED** [review]
LGTM. Can't formally approve since I had originally opened this PR.

**Comment by XedinUnknown** [comment]
Allow `CommandInvokerInterface#invoke()` to also use a `Traversable` for args.

**Comment by mecha** [comment]
LGTM 👍

**Pull request merged**